test/integration/src/test/java/io/pravega/test/integration/ControllerFailoverTest.java

```diff
@@ -145,4 +145,17 @@ public void testStop() throws Exception {
         controllerWrapper.awaitRunning();
         controllerWrapper.close();
     }
+
+    @Test(timeout = 180000)
```

Comment:
Does this test really take this long? 3 minutes is a very long time to add to our pre-submit for just this one test. 
Currently we run 900 tests in under 20 minutes, so adding 3 just for this would be a major loss. Can we find a way to make this faster?

Reply:
Once the SS is down and controller is shut down, in the eventProcessor there is a readeroffline api which is again trying to connect to SS with a finite number of retries, so there is the delay. I have tried to simulate the scenario using a unit test.